web/test/helpers_test.dart

```diff
@@ -36,4 +39,65 @@ void main() {
     // Ensure accessing any arbitrary item in the list does not throw.
     expect(() => dartList[0], returnsNormally);
   });
+
+  test('cross-origin windows and locations can be accessed safely', () {
```

Comment:
(optional) I wonder if it would be worth to also add test coverage of the unwrapped behavior?  That is, something that verifies that cross-origin windows/locations fail if you tried to access them directly?

Reply:
I think both the test runner and `package:test` have this issue. After a good amount of debugging, I can't tell how to enable same-origin policy. Filed https://github.com/dart-lang/test/issues/2282 for now and single-stepped to verify that there are no cross-origin issues with the current test.